tests/test_queryselector.py

```diff
@@ -0,0 +1,301 @@
+    page = isolated_page
+    elements = await page.Jx('/html/body/non-existing-element')
+    assert not elements and elements == []
+
+
+@chrome_only
+@sync
+async def test_Jx_returns_multiple_elements(isolated_page):
+    """Test Page().Jx() method should return multiple elements."""
+    page = isolated_page
+    await page.setContent('<div></div><div></div>')
+    elements = await page.Jx('/html/body/div')
+    assert len(elements) == 2
+
+
+@sync
+async def test_ElementHandle_J_queries_element(isolated_page, server):
+    """Test ElementHandle.J() method should query existing element."""
+    page = isolated_page
+    await page.goto(server / "playground.html")
+    await page.setContent('<html><body><div class="second"><div class="inner">A</div></div></body></html>')
+    html = await page.J('html')
+    second = await html.J('.second')
+    inner = await second.J('.inner')
+    content = await page.evaluate("e => e.textContent", inner)
+    assert content == 'A'
+
+
+@chrome_only
+@sync
+async def test_ElementHandle_J_returns_none_if_no_element(isolated_page):
+    """Test ElementHandle.J() method should return none for non existing element."""
+    page = isolated_page
+    await page.setContent('<html><body><div class="second"><div class="inner">B</div></div></body></html>')
+    html = await page.J('html')
+    second = await html.J('.third')
+    assert not second
+
+
+@chrome_only
+@sync
+async def test_ElementHandle_Jeval_evalates_js_func(isolated_page):
+    """Test ElementHandle.Jeval() method evaluates JS function against css selector."""
+    page = isolated_page
+    await page.setContent(
+        '<html><body><div class="tweet"><div class="like">100</div><div class="retweets">10</div></div></body></html>'
+    )
+    tweet = await page.J('.tweet')
+    content = await tweet.Jeval('.like', "node => node.innerText")
+    assert content == '100'
+
+
+@chrome_only
+@sync
+async def test_ElementHandle_Jeval_retrieves_content(isolated_page):
+    """Test ElementHandle.Jeval() method should retrieve content from subtree."""
+    page = isolated_page
+    htmlContent = '<div class="a">not-a-child-div</div><div id="myId"><div class="a">a-child-div</div></div>'
+    await page.setContent(htmlContent)
+    elementHandle = await page.J('#myId')
+    content = await elementHandle.Jeval('.a', "node => node.innerText")
+    assert content == 'a-child-div'
+
+
+@chrome_only
+@sync
+async def test_ElementHandle_Jeval_raises_exc_if_no_selector(isolated_page):
+    """Test ElementHandle.Jeval() throws exception in case of missing selector."""
+    page = isolated_page
+    htmlContent = '<div class="a">not-a-child-div</div><div id="myId"></div>'
+    await page.setContent(htmlContent)
+    elementHandle = await page.J('#myId')
+    with pytest.raises(ElementHandleError) as exc:
+        await elementHandle.Jeval('.a', "node => node.innerText")
+    assert 'Error: failed to find element matching selector ".a"' in str(exc)
```

Comment:
This is a mistake I made as well, but pytest.raises allows a match parameter (note that it takes a regex pattern):
```suggestion
    with pytest.raises(ElementHandleError, match='Error: failed to find element matching selector ".a"') as exc:
        await elementHandle.Jeval('.a', "node => node.innerText")
```
You can escape the dot if you wish and prepend `r` to the string, but I find this a little more readable even it is a bit more broad (eg it'll match stuff like `Error: failed to find element matching selector "ja"`)

Reply:
Fixed, thanks